Prolific/Prolific/View Controllers/Profile View Controller/ProfileViewController.m

```diff
@@ -153,6 +153,25 @@ - (void)uploadImage:(UIImage *)profileImage {
             NSLog(@"Error uploading data: %@", error.localizedDescription);
         }
     }];
+    
+    [uploadTask observeStatus:FIRStorageTaskStatusProgress
+                      handler:^(FIRStorageTaskSnapshot *snapshot) {
+        __strong typeof (weakSelf) strongSelf = weakSelf;
+        if (strongSelf == nil) return;
+        
+        strongSelf.uploadProgressBar.hidden = NO;
+        strongSelf.uploadProgressBar.progress = snapshot.progress.fractionCompleted;
+        NSLog(@"You are %f complete:", snapshot.progress.fractionCompleted);
+    }];
+    
+    [uploadTask observeStatus:FIRStorageTaskStatusSuccess
+                      handler:^(FIRStorageTaskSnapshot *snapshot) {
+        __strong typeof (weakSelf) strongSelf = weakSelf;
+        if (strongSelf == nil) return;
+        
+        strongSelf.uploadProgressBar.hidden = YES;
```

Comment:
so one interesting thing about weakself/strongself. Usually if your actual code is only one line you can just call `weakSelf.uploadProgressbar.hidden = YES`. 

If weakself is nil it won't do anything. You only need strongself when you have multiple actions. So let's say you have:

[
if (strongSelf == nil) return;
[strongSelf doSomething];
[strongSelf doSomethingElse];
](url)

Then you need strongSelf. If you used weakSelf, it could become nil between "doSomething" and "doSomethingElse"